Realm/RLMArray.mm

```diff
@@ -30,21 +30,55 @@
 @implementation RLMArray {
     // array for standalone
     NSMutableArray *_backingArray;
+    __weak RLMObjectBase *_parentObject;
 }
 
-- (instancetype)initWithObjectClassName:(NSString *)objectClassName standalone:(BOOL)standalone {
+static void changeArray(__unsafe_unretained RLMArray *const ar, NSKeyValueChange kind, NSIndexSet *is, dispatch_block_t f) {
+    if (!ar->_backingArray) {
+        ar->_backingArray = [NSMutableArray new];
+    }
+
+    if (RLMObjectBase *parent = ar->_parentObject) {
+        [parent willChange:kind valuesAtIndexes:is forKey:ar->_key];
+        f();
+        [parent didChange:kind valuesAtIndexes:is forKey:ar->_key];
+    }
+    else {
+        f();
+    }
+}
+
+static void changeArray(__unsafe_unretained RLMArray *const ar, NSKeyValueChange kind, NSUInteger index, dispatch_block_t f) {
+    changeArray(ar, kind, [NSIndexSet indexSetWithIndex:index], f);
+}
+
+static void changeArray(__unsafe_unretained RLMArray *const ar, NSKeyValueChange kind, NSRange index, dispatch_block_t f) {
+    changeArray(ar, kind, [NSIndexSet indexSetWithIndexesInRange:index], f);
+}
+
+- (instancetype)initWithObjectClassName:(NSString *)objectClassName
+                           parentObject:(id)object
+                                    key:(NSString *)key {
     self = [super init];
     if (self) {
         _objectClassName = objectClassName;
-        if (standalone) {
-            _backingArray = [[NSMutableArray alloc] init];
-        }
+        _parentObject = object;
+        _key = key;
     }
     return self;
 }
 
+// used by legacy models
 - (instancetype)initWithObjectClassName:(NSString *)objectClassName {
-    return [self initWithObjectClassName:objectClassName standalone:YES];
+    self = [super init];
```

Comment:
any reason not to call thru to `initWithObjectClassName:parentObject:key:` here?